Client/Experiments/Messaging/GleanPlumbMessageManager.swift

```diff
@@ -228,12 +227,18 @@ class GleanPlumbMessageManager: GleanPlumbMessageManagerProtocol {
             return nil
         }
 
+        let messageMetadata = messagingStore.getMessageMetadata(messageId: messageId)
+        if messageMetadata.impressions >= style.maxDisplayCount || messageMetadata.isExpired {
+            _ = messagingStore.onMessageExpired(messageMetadata, shouldReport: true)
+            return nil
+        }
+
```

Comment:
Aren't we doing the same "expired" check in `getAllValidMessagesFor(...)`?

`createMessage` was meant to handle everything about a message's data validity (handle a message's required properties, like triggers, action etc.). Expiration was intentionally left out and handled in later steps. 

But we can keep this here and remove the isExpired check in `getAllValidMessagesFor` if you want to do more.

Reply:
Oh I think I see - is this change to accommodate what's happened in the Store? Personally, I'd like keeping metadata updates in a single area like the Store. It's up to you

Reply:
I had to make this changes to accommodate to the collection view reload cycle and some issues we still have in home page. 
One of the issues is that we were counting the impression after the message was created so initially it will return the message as valid here than we will add the impression and the message will become not valid so in the next reload we will need to remove the card causing a crash on the app because of home page issues